File: Desktop/src/api/suppliers.ts
import { api } from './axios';
import type {
  Supplier,
  CreateSupplierInput,
  ListSuppliersParams,
} from '@/types/supplier';
import type { ApiPaginated } from '@/types/api';

export const supplierApi = {
  list: (params: ListSuppliersParams = {}) =>
    api
      .get<ApiPaginated<Supplier>>('/client/suppliers', { params })
      .then((r) => r.data),

  get: (id: string) =>
    api.get<{ data: Supplier }>(`/client/suppliers/${id}`).then((r) => r.data.data),

  create: (payload: CreateSupplierInput) =>
    api
      .post<{ data: Supplier }>('/client/suppliers', payload)
      .then((r) => r.data.data),

  update: (id: string, payload: Partial<CreateSupplierInput>) =>
    api
      .patch<{ data: Supplier }>(`/client/suppliers/${id}`, payload)
      .then((r) => r.data.data),

  remove: (id: string) =>
    api.delete(`/client/suppliers/${id}`).then((r) => r.data),

  orders: (id: string, params: { page?: number; limit?: number } = {}) =>
    api
      .get<ApiPaginated<unknown>>(`/client/suppliers/${id}/orders`, { params })
      .then((r) => r.data),
};

File: Android/client/lib/api/suppliers.dart
import 'api_client.dart';

class SuppliersApi {
  static final _api = BizOsApiClient.instance;

  static Future<List<Map<String, dynamic>>> list({
    Map<String, dynamic>? query,
  }) async => _api.list(await _api.get('/client/suppliers', query: query));
  static Future<Response<dynamic>> create(Map<String, dynamic> data) =>
      _api.post('/client/suppliers', data: data);
  static Future<Response<dynamic>> get(String id) =>
      _api.get('/client/suppliers/$id');
  static Future<Response<dynamic>> update(
    String id,
    Map<String, dynamic> data,
  ) => _api.patch('/client/suppliers/$id', data: data);
  static Future<Response<dynamic>> remove(String id) =>
      _api.delete('/client/suppliers/$id');
  static Future<Response<dynamic>> orders(
    String id, {
    Map<String, dynamic>? query,
  }) => _api.get('/client/suppliers/$id/orders', query: query);
}
